### PR TITLE
Use common docstrings between C++ and Python

### PR DESCRIPTION
**Pull request by haleepfl**

### CODE REVIEW

File: ci/check_generated_docstrings.sh [hunk 1]
```diff
@@ -6,19 +6,15 @@
 set -euo pipefail
 
 VENV=build/venv-docstrings
-CLANG_FORMAT_VERSION=9.0.0
+CLANG_VERSION=6.0.0.2
```
Comment (matz-e) [inline ci/check_generated_docstrings.sh:9]: :cold_face:

Comment (haleepfl) [reply ci/check_generated_docstrings.sh]: pybind11 actually has the required clang bindings as a submodule, so I've removed this

**Comment by haleepfl** [comment]
@matz-e and @mgeplf this is still a work in progress, but before I continue on, I would appreciate any feedback you have. That way I can address any comments as I transition remaining docstrings.

**Comment by mgeplf** [comment]
looks reasonable to me, so far

**Comment by matz-e** [comment]
Sorry, had this open and forgot about it. Looks OK to me!

…that generated macro expansion is just sick though. Makes anything I write in TeX look very sane.

**Comment by mgeplf** [comment]
Wow, again, tonnes of work, thanks.  So far everything looks reasonable, and I like the output I've seen.  Seems about as maintainable as one can expect too.  :+1:

**Comment by alkino** [comment]
I'm not sure we should push generated file to git, make it generated at compile time by cmake.

**Comment by haleepfl** [comment]
> I'm not sure we should push generated file to git, make it generated at compile time by cmake.

We debated this initially. Normally I would agree, however in this case the generation requires clang. Do we want to make that a required dependency just for people to compile it locally? Mitsuba (which also uses this approach) commits the docstring file to git: https://github.com/mitsuba-renderer/mitsuba2/blob/master/include/mitsuba/python/docstr.h

**Pull request merged**